homeassistant/components/ecowitt/sensor.py

```diff
@@ -215,11 +197,6 @@ def _new_sensor(sensor: EcoWittSensor) -> None:
         if sensor.stype not in ECOWITT_SENSORS_MAPPING:
             return
 
-        # Ignore metrics that are not supported by the user's locale
-        if sensor.stype in _METRIC and not hass.config.units.is_metric:
-            return
-        if sensor.stype in _IMPERIAL and hass.config.units.is_metric:
-            return
```

Comment:
We should add units that the user not need as disabled_by default. Now it generate mess

Reply:
Again ,this doesn't disable them as far as i can tell, this excludes them which is the problem I have as there is nothing to enable, if they had just been disabled by default and/or hidden we wouldn't be having this discussion nor would I be submitting a PR. 

You are also assuming what a user needs based on what you need, which in my case is neither of the above options provided. 

The reason for the mess is because the aioecowitt lib seems to have been written with the assumption of you are either imperial or metric and has used the same name for both values resulting in the duplication.

Happy to disable by default if you tell me how as given the structure of the code i cant see how. It has been greatly appreciated how you've taken the time to say 'i don't do for one country' and 'this is a mess' yet in probably less time you could have provided a code hint to help solve the problem.

Reply:
That's literally what Pascal is saying. You now remove the filtering, so it is always showing all sensors. Instead, use `entity_registry_enabled_default` dynamically using `replace` on the entity description classes to create new ones where `entity_registry_enabled_default=False` is set for entities that are not matching imperial/metric. 

We can't just add all sensors and give the user a mess. We allow imperial/metric in HA, so that's what we should go with. 

Alternatively, as Erik [mentioned](https://github.com/home-assistant/core/pull/79223#issuecomment-1261876237), new device classes have been added so existing sensors are customizable in the UI. That way a user can pick their own units without having to have entities with same data, just expressed in different units

Reply:
I like the replace idea, i'll maybe try that. In an ideal world the units prompt on first install would be nice to have a hybrid/mixed option